lesson/narrative_and_checkpoints/narrative-and-checkpoints-template.md

```diff
@@ -1,51 +1,417 @@
+```js
+//file: potassium_counter.test.js
+const potassiumCounter = require('./potassium_counter')
+ 
+test('successfully request potassium level and return in callback', () => {
+   //setup
+  const inputFood = null
+  const value = potassiumCounter.potassiumSearch(inputFood, result => {
+       //Jest test
+        expect(result).toBeDefined()
+      })
+})
+``` 
+Let’s see what happens when we try and get the potassium levels of ```null```:
+ 
+![false positive test output](assets/exercise_6_1.png)
+
+Notice how the test passes even though it shouldn't. There are a few ways to fix this issue. We will begin with the ```done()``` callback provided by Jest. When called, the ```done()``` callback signifies that the test should not finish until the asynchonous code has completed its behavior. The implementation of the ```done()``` callback to fix the false positive problem looks like this:
+
+```js
+test("successfully request potassium level and return in callback"), (done) => {
+   //setup
+  const inputFood = null
+  const value = potassiumCounter.potassiumSearch(inputFood, result => {
+       //Jest test
+       try {
+           expect(result).toBeDefined()
+           done()
+       } catch (error) {
+           done(error)
+       }    
+   })
+})
+```
+![false positive solution test output](assets/exercise_6_2.png)
+
+As you can see the test is now functioning as expected and will not pass until ```inputFood``` is passed defined as a valid argument for the ```potassiumSearch()``` function.
+
+
+
+### Instructions:
 
-Hint: _Insert optional but recommended hint text here._
+1. Checkpoint: _Run test to identify if it is returning a  false positive_
 
-2. Checkpoint: _Insert checkpoint text here._
+    Hint: _Run ```npm test```_
+
+2. Checkpoint: _Using the example above as a reference rewrite the asynchronous test using the `done()` callback._
+
+    Hint: _ ```done``` should be passed as a parameter and the test syntax should include:
+```js
+        try {
+        expect(result).toBeDefined()
+        done()
+        } catch (error) {
+            done(error)
+        }
+```
+
+3. Checkpoint: _Run test to verify that the tests do not pass and that the false positive problem has been fixed_
+
+    Hint: _Run ```npm test```_
+
+## Exercise 7: _Testing Async Code in Jest (Part 2)_
+
+### Narrative:
+
+As mentioned before, there are different ways beyond the ```done()``` callback to properly perform tests on asynchronous code. We will now explore testing functions that return a promise instead of a callback function. Let’s use the same function as before altering it slightly. Instead of returning a callback with the data, the function will now simply return a promise. We will run the same test as before but this time without the done parameter to recreate the false positive problem:
+```js
+test('successfully request potassium level from database and return as promise', () => {
```

Comment:
These tests are describing implementation details now. Tests should avoid caring about how a unit is implemented. The only thing the test cares about is that it got the potassium level.